backend/dummy.go

```diff
@@ -36,167 +35,200 @@ type OrderInfo struct {
 
-		timestamp, err := time.Parse("01/02 03:04:05PM '06 -0700", month+"/"+day+" 00:00:00AM '"+year[2:]+" -0000")
+		//now we just parse the float, removing whitespace and a "$" from the string
+		itemOriginalPrice, err := strconv.ParseFloat(strings.TrimSpace(record[ORIGINAL_PRICE_COLUMN][1:]), 64)
 		if err != nil {
-			fmt.Println(err)
+			return nil, errors.New("parseCSV: could not parse original price" + err.Error())
 		}
 
-		if time.Duration.Hours(time.Now().Sub(timestamp)) > 24*30 {
-			//fmt.Println("over 30 days")
-			continue
-		}
+		//we can now mostly make an OrderInfo object. Unassigned fields will have value of -1
+		orderHistory = append(orderHistory, OrderInfo{
+			Name:          itemName,
+			DateOrdered:   itemDateOrdered,
+			Asin:          itemAsin,
+			OriginalPrice: itemOriginalPrice,
+			CurrentPrice:  -1,
+			PriceDrop:     -1})
+	}
 
-		//populate our order info fields and create struct
-		s0, s1, s2 := record[2], record[0], record[4]
+	return orderHistory, nil
+}
 
-		//now we just parse the float, removing any whitespace from the string
-		f1, err := strconv.ParseFloat(strings.TrimSpace(record[12][1:]), 64)
-		if err != nil {
-			fmt.Println(err)
-		}
+func isItemOrderedWithin30Days(orderDate string) (bool, error) {
+	//parsing date to see if item is with 30 day return window or not
+	timeStringArr := strings.Split(orderDate, "/")
+	if len(timeStringArr[0]) == 1 {
+		timeStringArr[0] = "0" + timeStringArr[0]
+	}
+	if len(timeStringArr[1]) == 1 {
+		timeStringArr[1] = "0" + timeStringArr[1]
+	}
+	timeStringArr[2] = timeStringArr[2][2:]
+	formattedTime := timeStringArr[0] + "/" + timeStringArr[1] + "/" + timeStringArr[2]
 
-		//we can now mostly make an OrderInfo object
-		orderhist = append(orderhist, OrderInfo{s0, s1, s2, f1, 0, 0})
+	timestamp, err := time.Parse("01/02/03", formattedTime)
+	if err != nil {
+		return false, errors.New("parseCSV: " + err.Error())
 	}
 
-	return orderhist
+	//this if statement will activate when the item is within 30 days of being ordered (in other words, the item is in the return window.)
+	if time.Duration.Hours(time.Now().Sub(timestamp)) > 24*30 {
+		return false, nil
+	}
+	return true, nil
 }
 
-// populates currente price and price drop in orderhist array
-// uses sync.WaitGroup
-func getoriginalprice(orderhist *[]OrderInfo) {
-	var wg sync.WaitGroup
-	for i := range *orderhist {
-		wg.Add(1) //we are starting a goroutine
-		go getOriginalPricePerItem(&((*orderhist)[i]), &wg)
+// populates currente price and price drop in orderHistory slice
+func getPriceInfo(orderHistory *[]OrderInfo) error {
+	eg := new(errgroup.Group)
+	for _, item := range *orderHistory {
+		//we are starting a new goroutine
+		eg.Go(func() error {
+			return getPriceInfoForItem(&item)
+		})
+	}
+	//the first goroutine that returns a non-nil error will output an error, otherwise we wait until all goroutines are finished.
+	if err := eg.Wait(); err != nil {
+		return errors.New("getPriceInfo: " + err.Error())

```

Comment:
idk exactly how all of this works, but if one or more price fetch fails, but others succeed, make sure you don't return early, let's try to get all of the prices